sources/packages/backend/apps/api/src/services/education-program/education-program.service.ts

```diff
@@ -268,16 +268,14 @@ export class EducationProgramService extends RecordDataModelService<EducationPro
     // For getting total raw count before pagination.
     const sqlQuery = paginatedProgramQuery.getSql();
 
-    if (paginationOptions.pageLimit) {
+    if (paginationOptions.pageLimit > 0) {
       paginatedProgramQuery.limit(paginationOptions.pageLimit);
     }
-    if (paginationOptions.page) {
-      paginatedProgramQuery.offset(
-        paginationOptions.page * paginationOptions.pageLimit,
-      );
-    } else {
-      paginatedProgramQuery.limit(0);
-    }
+
+    paginatedProgramQuery.offset(
+      paginationOptions.page * paginationOptions.pageLimit,
```

Comment:
what happens here when `paginationOptions.page` is not set (no pagination used)?

Reply:
what do you mean by no pagination? pagination will be always there, right?

Reply:
Because all the pagination properties are query parameters and technically the API can be used without pagination and sort.

Reply:
i think pagination will be always there, i can't see a scenario, where there is no pagination

Reply:
On `PaginationOptionsAPIInDTO`, `page` and `pageLimit` are required. I believe the methods that are built for pagination must provide the pagination parameters.